packages/base/src/3dview/mainview.tsx

```diff
@@ -1019,7 +1021,16 @@ export class MainView extends React.Component<IProps, IStates> {
     return mesh;
   }
 
+  private _previousSelection: { [key: string]: ISelection } | null = null;
   private _updateSelected(selection: { [key: string]: ISelection }) {
+    const selectionChanged =
+      JSON.stringify(selection) !== JSON.stringify(this._previousSelection);
+
+    if (!selectionChanged) {
+      return;
+    }
+    this._previousSelection = { ...selection };
```

Comment:
who do we need this logic? If we really need to filter out this case, maybe it's more appropriate to do it in the `_onClick` method and use `_selectedMeshes` property.

Reply:
It is probably good to have a watchdog here to prevent going further updating the selection when it has not changed.

However, it may be good to track what it calling this so many times. I wonder if it could be a tiny bit related to #560

Reply:
> I wonder if it could be a tiny bit related to #560

Yes it can most likely be the case looking at the frequency selection is changing in #560, though can't be said with full certainty since we don't have a definite way to reproduce it yet

> it may be good to track what is calling this so many times.

Yes, i'll be on it

Reply:
Thanks

Reply:
It is happening here:

https://github.com/jupytercad/JupyterCAD/blob/4f7c1fc5752624155257d9a74d8ebb5fa7d721fe/packages/base/src/3dview/mainview.tsx#L1179-L1181